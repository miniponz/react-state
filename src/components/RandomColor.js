import React, { PureComponent } from 'react';

export default class RandomColor extends PureComponent {
  state = {
    color: ''
  }

  randomColor = () => {
    const colors = ['red', 'orange', 'yellow', 'green', 'blue', 'indigo', 'violet'];
    const num = Math.floor(Math.random() * colors.length);
    return colors[num];
  }

  componentDidMount() {
    this.intervalId = setInterval(() => {
      this.setState({ color: this.randomColor() });
      console.log(this.state);
    }, 1000);
  }

  render() {
    const { color } = this.state;
    return (
      <>
      <h1>RANDOM COLORS</h1>
      <div style={ { background: color, height: '100px', width: '100px' } }></div>
      </>
    );
  }
}



