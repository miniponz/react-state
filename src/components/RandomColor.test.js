import React from 'react';
import { shallow } from 'enzyme';
import RandomColor from './RandomColor';

describe('color picker', () => {
  it('renders a color picker', () => {
    const wrapper = shallow(<RandomColor />);
    expect(wrapper).toMatchSnapshot();
  });

  
});
