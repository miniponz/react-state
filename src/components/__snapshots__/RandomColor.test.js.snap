// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`color picker renders a color picker 1`] = `
<Fragment>
  <div
    style={
      Object {
        "background": "",
        "height": "100px",
        "width": "100px",
      }
    }
  />
</Fragment>
`;
